Assets/Scripts/Guests/Guest.cs

```diff
@@ -109,25 +161,134 @@ private void ChangeBehaviour()
+	/// </summary>
+	/// <returns>All tiles neighbouring in the cardinal directions</returns>
+	private List<Vector2> FindPossibleTilesToMoveTo()
+	{
+		Vector2 currentOffset = FindCurrentOffset();
+		List<Vector2> possibleOffsets = new List<Vector2>();
+
+		foreach (Vector2 v in currentRoom.roomShape.OffsetFromRoomCenter)
+		{
+
+			if (Vector2.Distance(v, currentOffset) == 1f)
+			{
+				possibleOffsets.Add(v);
+			}
+		}
+
+		return possibleOffsets;
+	}
+
+	/// <summary>
+	/// Forces to leave the hotel
+	/// </summary>
+	public void ForceLeave()
+	{
+		enjoyTimeLeft = 0f;
+		numOfRoomsToVisit = 0;
+		BeginLeaving();
+
+	}
+
+
+	/// <summary>
+	/// Is the guest at the door?
+	/// </summary>
+	/// <returns>boolean that displays if the guest is on the door's position</returns>
+	private bool IsAtRoomDoor()
+	{
+		if (currentRoom == null) return true;
+
+		return (Vector2) transform.position == (Vector2) currentRoom.transform.position + currentRoom.doorOffset;
+	}
+
+	/// <summary>
+	/// Moves around in the room, looking for a random neighbouring spot
+	/// </summary>
+	private void MoveToNeighbouringTile()
+	{
+		List<Vector2> possibleOffsets = FindPossibleTilesToMoveTo();
+
+		if (possibleOffsets.Count == 0)
+		{
+			moving = false;
+		}
+		else
+		{
+			target = (Vector2) currentRoom.transform.position + possibleOffsets[Random.Range(0, possibleOffsets.Count)];
+			moving = true;
+		}
+	}
+
+	/// <summary>
+	/// Finds the next tile to move towards 
+	/// </summary>
+	private void PathAndMoveTowardsTile(Vector2 targetOffset)
+	{
+		if(shortestPath == null)
+		{
+			Vector2 myOff = FindCurrentOffset();
+			Debug.Log("find path from "+myOff+" to "+targetOffset);
+			shortestPath = currentRoom.GetShortestPath(myOff, targetOffset);
+			Debug.Log("path = "+string.Join(",", shortestPath.ToArray()));
+		}
+		// needs to have at least 2 targets in it, otherwise reached the end
+		if(shortestPath.Count < 2)
+		{
+			moving = false;
+			return;
+		}
+		// remove the reached target, and set the next
+		shortestPath.RemoveAt(0);
+		target = (Vector2) currentRoom.transform.position + shortestPath[0];
+		moving = true;
+	}
+
 	/// <summary>
 	/// Tries to find a room to stay in, returns if it is successful
 	/// </summary>
 	private void TryFindRoom()
-	{
+	{		
+
+		if(currentPathfindCooldown > 0f)
+		{
+			currentPathfindCooldown -= Time.deltaTime;
+			return;
+		}
+
 		HotelRoom roomToVisit = null;
 
-		List<HotelRoom> visitableRooms = MapManager.hotelRooms.FindAll(r => !r.Flooded && !r.Sunk && r != currentRoom);
+		List<HotelRoom> visitableRooms = MapManager.hotelRooms.FindAll(r => !r.Flooded && !r.Sunk && r != currentRoom
+		&& !r.AtCapacity);
 
 		if(visitableRooms.Count > 0)
 		{
-			int random = Random.Range(0, visitableRooms.Count);
-			roomToVisit = visitableRooms[random];
+			int randomRoomIndex = Random.Range(0, visitableRooms.Count);
+			roomToVisit = visitableRooms[randomRoomIndex];
+		}
+		else
+		{
+			currentPathfindCooldown = pathfindCooldown;

```

Comment:
you could just early return here
```suggestion
			currentPathfindCooldown = pathfindCooldown;
			return;
```

Reply:
better to return the earliest
consider rewrite the function like this:
```cs
{
  visitableRooms = FindAll...
  if(visitableRooms.Count == 0)
  {
    currentPathfindCooldown = pathfindCooldown;
    return;
  }
  roomToVisit = visitableRooms[randomIndex];
  // all the other code processing the selected room
}
```

Reply:
I'll take a look